subsys/profiler/profiler_nordic.c

```diff
@@ -246,17 +246,48 @@ void profiler_log_start(struct log_event_buf *buf)
 	/* Adding one to pointer to make space for event type ID */
 	__ASSERT_NO_MSG(sizeof(uint8_t) <= CONFIG_PROFILER_CUSTOM_EVENT_BUF_LEN);
 	buf->payload = buf->payload_start + sizeof(uint8_t);
-	profiler_log_encode_u32(buf, k_cycle_get_32());
+	profiler_log_encode_uint32(buf, k_cycle_get_32());
 }
 
-void profiler_log_encode_u32(struct log_event_buf *buf, uint32_t data)
+void profiler_log_encode_uint32(struct log_event_buf *buf, const uint32_t data)
 {
 	__ASSERT_NO_MSG(buf->payload - buf->payload_start + sizeof(data)
 			 <= CONFIG_PROFILER_CUSTOM_EVENT_BUF_LEN);
 	sys_put_le32(data, buf->payload);
 	buf->payload += sizeof(data);
 }
 
+void profiler_log_encode_int32(struct log_event_buf *buf, const int32_t data)
+{
+	profiler_log_encode_uint32(buf, (const uint32_t)data);
+}
+
+void profiler_log_encode_uint16(struct log_event_buf *buf, const uint16_t data)
+{
+	__ASSERT_NO_MSG(buf->payload - buf->payload_start + sizeof(data)
+			 <= CONFIG_PROFILER_CUSTOM_EVENT_BUF_LEN);
+	sys_put_le16(data, buf->payload);
+	buf->payload += sizeof(data);
+}
+
+void profiler_log_encode_int16(struct log_event_buf *buf, const int16_t data)
+{
+	profiler_log_encode_uint16(buf, (const uint16_t)data);
+}
+
+void profiler_log_encode_uint8(struct log_event_buf *buf, const uint8_t data)
+{
+	__ASSERT_NO_MSG(buf->payload - buf->payload_start + sizeof(data)
+			 <= CONFIG_PROFILER_CUSTOM_EVENT_BUF_LEN);
+	*(buf->payload) = data;
+	buf->payload += sizeof(data);
+}
+
+void profiler_log_encode_int8(struct log_event_buf *buf, const int8_t data)
+{
+	profiler_log_encode_uint8(buf, (const uint8_t)data);
```

Comment:
casting to `const` is not needed; note that we don't do it when passing pointers :)